### PR TITLE
[FIX] mrp_subcontracting_purchase: fix received_qty with return move

### PR DESCRIPTION
**Pull request by DavidFesquet**
method _is_purchase_return had a complex check intended to detect subcontract return move. This Check would incorrectly pass for subcontract move when the destination location (input) does not belong to the customer warehouse.

To fix this issue, we can check the field StockMove.is_subcontract, however this field is only available if mrp_subcontracting is installed. Furthermore, the check itself make no sens in purchase_stock if the module mrp_subcontracting is not install Hence, we can move this check to mrp_subcontracting_purchase.

# To Reproduce:
With Purchase & Mrp Subcontracting installed:
- Create Component C, consumable
- Create Product P, storable, Set Vendor V under Purchase Tab
- Create BOM for P, subcontracted with Vendor V, and C as component
- In the Warehouse, set 2 steps reception
- Set Input parent location to 'Physical Location'
- Create PO to vendor V, product P, confirm, receive Product => Received Qty for P in PO show 0

OPW-3216011

---

BEFORE:
```python
self.location_dest_id.usage == "supplier" or (
                self.location_dest_id.usage == "internal"
                and self.location_id.usage != "supplier"
                and self.warehouse_id
                and self.location_dest_id not in self.env["stock.location"].search([("id", "child_of", self.warehouse_id.view_location_id.id)])
```
The big chunk of test is to handle subcontract moves (https://github.com/odoo/odoo/commit/e6a1e240f18bbc95475cd82154bb91a455704b15).

So I did 2 things:
- Rewrote this big chunk
- Moved it to mrp_subcontracting under the name of `_is_subcontract_return`
- Join the 2 check in `mrp_subcontracting_purchase` 

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp_subcontracting/models/stock_move.py [hunk 1]
```diff
@@ -239,3 +239,9 @@ def _update_subcontract_order_qty(self, new_quantity):
                         'mo_id': production.id,
                         'product_qty': production.product_uom_qty - quantity_to_remove
                     }).change_prod_qty()
+
+    def _is_subcontract_return(self):
+        self.ensure_one()
+        origin_is_subcontract = self.move_orig_ids and all(self.move_orig_ids.mapped('is_subcontract'))
```
Comment (Whenrow) [inline addons/mrp_subcontracting/models/stock_move.py:245]: I would use returned_move_id and origin_returned_move_id keys to get the connection between the subcontract move and the return move.

Comment (DavidFesquet) [reply addons/mrp_subcontracting/models/stock_move.py:245]: Nice, didn't know it existed

**Comment by Whenrow** [comment]
robodoo r+

**Comment by Whenrow** [comment]
robodoo rebase-ff

**Pull request closed without merging**